### DISCUSSION TITLE
PandasTools is not catching the substructure when includeFingerprints=True

### DISCUSSION THREAD

**Discussion by lhyuen** [rdkit/rdkit#5476] [General]
Hello rdkit community, 

I am still quite new to rdkit and I am not sure if this is a bug or not so I want to start the discussion instead of directly going to the bug report. I recently tried to perform some substructure filtering from a dataframe using the PandasTools and ran into an unexpected behavior where it would only work when includeFingerprints = False. 

Here are some steps to reproduce the behavior:

import pandas as pd
from rdkit.Chem import PandasTools
from rdkit import Chem

#I would like to catch that aromatic carbon labelled as [c;R] in the SMARTS pattern
```
pattern=Chem.MolFromSmarts('[n]1:[n]:[c]:[c]2:[c](-[c;R]):[c]:[c]:[c](:[c]:12)')

test=pd.DataFrame(['C1(C2=CC=CC3=C2C=NN3)=CC=CC=C1'], columns=['smiles'])
PandasTools.AddMoleculeColumnToFrame(test, 'smiles', includeFingerprints=False)
test.loc[test['ROMol']>=pattern][['ROMol']] #-> this would work fine

PandasTools.AddMoleculeColumnToFrame(test, 'smiles', includeFingerprints=True)
test.loc[test['ROMol']>=pattern][['ROMol']] #-> this would give me an error saying that non-ring atom 5 marked aromatic
```

Is it something wrong with my SMARTS pattern? 

I am using rdkit  '2022.03.4' in a linux environment. I am happy to provide further information as needed.

**Comment by greglandrum** [comment]
@lhyuen sorry this went unreplied to for so long.
The sample code you provided doesn't work: the SMARTS that you use to define pattern will not parse since it's missing some ring closure digits:
```
[04:54:10] SMARTS Parse Error: unclosed ring for input: '[n]1:[n]:[c]:[c]2:c:[c]:[c]:c'
```
If you are still having this problem, please fix the SMARTS to be what you intended and then I can take a look at this.

**Comment by lhyuen** [comment]
@greglandrum Sorry for the confusion. Because I forgot to post the code within a codeblock using markdown, the SMARTS string came out weird. I have fixed the original post with proper formatting.

The SMARTS I provided was this:

```
pattern=Chem.MolFromSmarts('[n]1:[n]:[c]:[c]2:[c](-[c;R]):[c]:[c]:[c](:[c]:12)')
```

Thank you for looking into it!
